### PR TITLE
Update bazel_compdb to: 0.5.2 (current: 0.4.5)

### LINKED ISSUE #19153: Newer release available `bazel_compdb`: 0.5.0 (current: 0.4.5)
Package Name: bazel_compdb
Current Version: 0.4.5@2020-08-01 07:23:27
Available Version: 0.5.0@2021-09-09 12:24:05
Upstream releases: https://github.com/grailbio/bazel-compilation-database/releases

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

Commit Message: Update bazel_compdb to: 0.5.2 (current: 0.4.5)
Additional Description:

Fixes #19153

### CODE REVIEW

**Comment by kfaseela** [comment]
cc @phlax

**Review by htuch — DISMISSED** [review]
/lgtm deps

File: tools/gen_compilation_database.py [hunk 1]
```diff
@@ -25,11 +25,21 @@ def generate_compilation_database(args):
...
 
-    compdb = []
-    for compdb_file in Path(execroot).glob("**/*.compile_commands.json"):
-        compdb.extend(
-            json.loads("[" + compdb_file.read_text().replace("__EXEC_ROOT__", execroot) + "]"))
-    return compdb
+    db_entries = []
+    for db in pathlib.Path(execroot).glob('**/*.compile_commands.json'):
+        with open(db, 'r') as f:
+            db_entries.extend(json.load(f))
```
Comment (phlax) [inline tools/gen_compilation_database.py:31]: probs just use pathlib here - ie 

```python
for db in pathlib.Path(execroot).glob('**/*.compile_commands.json'):
        db_entries.extend(json.loads(db.read_text()))
```

File: tools/gen_compilation_database.py [hunk 2]
```diff
@@ -25,11 +25,21 @@ def generate_compilation_database(args):
...
+
+    def replace_execroot_marker(db_entry):
+        if 'directory' in db_entry and db_entry['directory'] == '__EXEC_ROOT__':
+            db_entry['directory'] = execroot
+        if 'command' in db_entry:
+            db_entry['command'] = (
+                db_entry['command'].replace('-isysroot __BAZEL_XCODE_SDKROOT__', ''))
+        return db_entry
+
+    db_entries = list(map(replace_execroot_marker, db_entries))
```
Comment (phlax) [inline tools/gen_compilation_database.py:41]: i think just return the `list(map(re...` rather than creating a var

**Review by phlax — CHANGES_REQUESTED** [review]
looking good @kfaseela thanks for taking the time to work through this

ive left a couple of nits inline...

File: tools/gen_compilation_database.py (hunk 2 above)
Comment (kfaseela) [reply tools/gen_compilation_database.py]: Done

**Review by phlax — APPROVED** [review]
lgtm, thanks for figuring this all out @kfaseela

**Pull request merged**